pyproject.toml

```
@@ -92,15 +94,29 @@ log_cli_level = "INFO"
 xfail_strict = true
 filterwarnings = [
     "error",
-    "ignore:.*pkg_resources.*:DeprecationWarning:",
-    "ignore:.*sre_.*:DeprecationWarning:",
-    "ignore:.*Rigetti.*:UserWarning:",
-    "ignore:.*Values in x.*:RuntimeWarning:",
-    "ignore:.*divide by zero encountered in det.*:RuntimeWarning:",
-    "ignore:.*invalid value encountered in det.*:RuntimeWarning:",
-    "ignore::DeprecationWarning:.*(docplex).*",
-    "ignore::SyntaxWarning:.*(docplex).*",
-    "ignore:.*The abstract Provider and ProviderV1 classes are deprecated.*:DeprecationWarning:qiskit.*",
+    'ignore:.*pkg_resources.*:DeprecationWarning:',
+    'ignore:.*sre_.*:DeprecationWarning:',
+    'ignore:.*Rigetti.*:UserWarning:',
+    'ignore:.*Values in x.*:RuntimeWarning:',
+    'ignore:.*divide by zero encountered in det.*:RuntimeWarning:',
+    'ignore:.*invalid value encountered in det.*:RuntimeWarning:',
+    'ignore::DeprecationWarning:.*(docplex).*',
+    'ignore::SyntaxWarning:.*(docplex).*',
+    'ignore:.*The abstract Provider and ProviderV1 classes are deprecated.*:DeprecationWarning:qiskit.*',
+    'ignore:.*qiskit.providers.models is deprecated since Qiskit 1.2*:DeprecationWarning:',
+    'ignore:.*Treating CircuitInstruction as an iterable is deprecated legacy behavior since Qiskit 1.2*:DeprecationWarning:pytket.*',
+    'ignore:.*The class ``qiskit.providers.models.backendconfiguration.GateConfig`` is deprecated as of qiskit 1.2.*:DeprecationWarning:qiskit.*',
+    'ignore:.*The class ``qiskit.providers.models.backendconfiguration.BackendConfiguration`` is deprecated as of qiskit 1.2.*:DeprecationWarning:qiskit.*',
+    'ignore:.*The class ``qiskit.providers.models.backendconfiguration.QasmBackendConfiguration`` is deprecated as of qiskit 1.2.*:DeprecationWarning:qiskit.*',
+    'ignore:.*The class ``qiskit.providers.models.backendproperties.BackendProperties`` is deprecated as of qiskit 1.2.*:DeprecationWarning:qiskit.*',
```

```suggestion
    'ignore:.*qiskit.providers.models.* is deprecated since Qiskit 1.2*:DeprecationWarning:',
```
Fairly sure that something like the suggestion should work to unify all these ignores.
Something similar can definitely be done for the `qiskit.primitives.*` warnings.

---

Done, but did it slightly differently and therefore did not commit the suggestion itself.